formapp/app.py

```diff
@@ -173,21 +173,21 @@ def newStudent(remote, username, bot=False):
             'excluded': False,
             'preferredMode': None
         }
-        #Track if this is the first human login or not
-        humanUsers = 0
+        #Track if the teacher is logged in
+        teacher = False
 
-        #Check each student so far to make sure that none of them are bots
+        #Check each student so far to make sure that none of them have teacher perms
         for user in sD.studentDict:
-            if not sD.studentDict[user]['bot']:
-                humanUsers += 1
+            if sD.studentDict[user]['perms'] == 0:
+                teacher = True
 
         #Login bots as guest
         if bot:
             print("[info] " +"Bot successful login. Made them a guest: " + username)
             sD.studentDict[remote]['perms'] = sD.settings['perms']['anyone']
 
-        #Login first user as teacher
-        elif humanUsers == 1:
+        #Login as teacher if there is no teacher yet
+        elif not teacher:
```

Comment:
What happens if a bot logs in first, such as bot_ir.py?

Reply:
It still won't make a bot the teacher. When someone logs in, it now looks for a teacher instead of any human user.